ci/da-ghc-lib/compile.yml

```diff
@@ -41,15 +42,20 @@ jobs:
       git clone https://github.com/digital-asset/ghc-lib.git
       cd ghc-lib
       git checkout $(ghc-lib-sha)
-      stack runhaskell --package extra \
+      mkdir -p ~/.stack
+      echo -e "system-ghc: true\ninstall-ghc: false" > ~/.stack/config.yaml
+      cat ~/.stack/config.yaml
+      STACK_CMD="stack runhaskell --package extra \
                        --package optparse-applicative \
                        CI.hs \
                        -- \
                          --da \
                          --merge-base-sha=$(base-sha) \
                          $(for p in $(patches); do echo -n "--patch $p "; done) \
                          --gen-flavor=$(flavor) \
-            | tee log
+            | tee log"
+      export NIX_PATH=nixpkgs=https://github.com/NixOS/nixpkgs/archive/cc855b133100a956d4587079e27c5704cbc8342b.tar.gz
+      nix-shell -p stack -p gmp -p autoconf269 -p automake -p coreutils -p perl -p m4 -p 'haskell.packages.ghc8104.ghcWithPackages(_: [])' 'haskell.packages.ghc865Binary.ghcWithPackages(_: [])' --command "$STACK_CMD"
```

Comment:
Why do we have three different GHC versions involved here? Maybe worth adding a comment.
```
flavor: 'ghc-8.8.1'
-p 'haskell.packages.ghc8104.ghcWithPackages(_: [])' 'haskell.packages.ghc865Binary.ghcWithPackages(_: [])'
```

Reply:
added a comment. 8.10.4 is what we use to build & run ghc-lib-gen. 8.6.5 is what we use to build our GHC fork based on 8.8.1.

Reply:
Thanks!